promcda/utils/utils_for_main.py

```diff
@@ -1055,16 +819,65 @@ def run_mcda_with_indicator_uncertainty(input_config: dict, input_matrix: pd.Dat
                           iterative_random_w_score_means=None,
                           iterative_random_w_score_stds=None,
                           iterative_random_w_score_means_normalized=None,
-                          input_matrix=input_matrix, config=input_config,
+                          input_matrix=input_matrix, config=extracted_values,
                           is_robustness_indicators=is_robustness_indicators)
 
 
-def _compute_scores_for_all_random_weights(indicators: dict, is_sensitivity: str,
-                                           weights: Union[List[str], List[pd.DataFrame], dict, None],
-                                           f_agg: str) -> tuple[Any, Any, Any, Any]:
+def check_input_matrix(input_matrix: pd.DataFrame) -> pd.DataFrame:
+    """
+    Check the input matrix for duplicated rows in the alternatives column, rescale negative indicator values
+    and drop the index column of alternatives.
+
+    Parameters:
+    - input_matrix: The input matrix containing the alternatives and indicators.
+
+    Raises:
+    - ValueError: If duplicated rows are found in the alternative column.
+    - UserStoppedInfo: If the user chooses to stop when duplicates are found.
+
+     :param input_matrix: pd.DataFrame
+     :rtype: pd.DataFrame
+     :return: input_matrix
+    """
+    if input_matrix.duplicated().any():
+        raise ValueError('Error: Duplicated rows in the alternatives column.')
+    elif input_matrix.iloc[:, 0].duplicated().any():
+        logger.info('Duplicated rows in the alternatives column.')
+
+    index_column_values = input_matrix.index.tolist()
+    logger.info("Alternatives are {}".format(index_column_values))
+    input_matrix_no_alternatives = input_matrix.reset_index(drop=True)  # drop the alternative
```

Comment:
`input_matrix_no_alternatives` variable name is confusing to me. `no` could mean `NO` or `number` so in my opinion a better name would be `input_matrix_without_alternatives`. This also applies to ProMCDA's `self.input_matrix_no_alternatives`.

An improvement would be to combine last 3 lines of this method into a single return statement. 

```
return _check_and_rescale_negative_indicators(input_matrix.reset_index(drop=True))
```